packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -316,17 +318,32 @@ class Tabs extends Foundation<TabsHandledProps, TabsUnhandledProps, TabsState> {
      * Handles the keydown event on the tab element
      */
     private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        const isHorizontal: boolean = this.props.orientation === Orientation.horizontal;
+
         switch (e.keyCode) {
             case keyCodeArrowLeft:
+                if (isHorizontal) {
+                    e.preventDefault();
+                    this.activateTab(TabLocation.previous);
+                }
+                break;
             case keyCodeArrowUp:
-                e.preventDefault();
-                this.activateTab(TabLocation.previous);
+                if (!isHorizontal) {
+                    e.preventDefault();
+                    this.activateTab(TabLocation.previous);
+                }
                 break;
             case keyCodeArrowRight:
+                if (isHorizontal) {
+                    e.preventDefault();
+                    this.activateTab(TabLocation.next);
+                }
+                break;
             case keyCodeArrowDown:
-                e.preventDefault();
-
-                this.activateTab(TabLocation.next);
+                if (!isHorizontal) {
+                    e.preventDefault();
+                    this.activateTab(TabLocation.next);
+                }
```

Comment:
I wonder if it might be easier to have an abstraction more like this:
```typescript
// horizontal orientation utilizes the left and right arrows
// while vertical orientation utilizes the up and down arrows
if (this.props.orientation === Orientation.horizontal) {
    switch (e.keyCode) {
        case keyCodeArrowLeft:
            e.preventDefault();
            this.activateTab(TabLocation.previous);
            return;
        case keyCodeArrowRight:
            e.preventDefault();
            this.activateTab(TabLocation.next);
            return;
    }
} else {
    switch (e.keyCode) {
        case keyCodeArrowUp:
            e.preventDefault();
            this.activateTab(TabLocation.previous);
            return;
        case keyCodeArrowDown:
            e.preventDefault();
            this.activateTab(TabLocation.next);
            return;
    }
}

switch (e.keyCode) {
    case keyCodeHome:
        this.activeTab(TabLocation.first);
        return;
    case keyCodeEnd:
        this.activeTab(TabLocation.last);
        return;
}
```

Reply:
sure, good suggestion. Updated.